src/ArmTemplates/Extractor/Models/ExtractorParameters.cs

```diff
@@ -1,56 +1,133 @@
-        public serviceUrlProperty[] serviceUrlParameters { get; set; }
-        public bool paramServiceUrl { get; private set; }
-        public bool paramNamedValue { get; private set; }
-        public bool paramApiLoggerId { get; private set; }
-        public bool paramLogResourceId { get; private set; }
-        public bool notIncludeNamedValue { get; private set; }
-        public bool paramNamedValuesKeyVaultSecrets { get; private set; }
-
-        public int operationBatchSize { get; private set; }
-
-        public bool paramBackend { get; set; }
-
-        public ExtractorParameters(ExtractorConfig exc, string dirName)
+        public string SourceApimName { get; private set; }
+
+        public string DestinationApimName { get; private set; }
+
+        public string ResourceGroup { get; private set; }
+
+        /// <summary>
+        /// Naming of all templates from the base folder requested by end-user in extractor-configuration parameters
+        /// </summary>
+        public FileNames FileNames { get; private set; }
+
+        /// <summary>
+        /// Not changeable root directory, where templates will be generated.
+        /// </summary>
+        public string FilesGenerationRootDirectory { get; private set; }
+
+        /// <summary>
+        /// Name of a single API that user wants to extract
+        /// </summary>
+        public string SingleApiName { get; set; }
+
+        /// <summary>
+        /// Names of APIs that user wants to extract
+        /// </summary>
+        public List<string> MultipleApiNames { get; set; }
+
+        /// <summary>
+        /// Create split api templates for every API in the source APIM instance
+        /// </summary>
+        public bool SplitApis { get; private set; }
+
+        /// <summary>
+        /// Are revisions included to templates
+        /// </summary>
+        public bool IncludeAllRevisions { get; private set; }
+
+        public string LinkedTemplatesBaseUrl { get; private set; }
+
+        public string LinkedTemplatesSasToken { get; private set; }
+
+        public string LinkedTemplatesUrlQueryString { get; private set; }
+
+        public string PolicyXMLBaseUrl { get; private set; }
+
+        public string PolicyXMLSasToken { get; private set; }
+
+        public string ApiVersionSetName { get; private set; }
+
+        public ServiceUrlProperty[] ServiceUrlParameters { get; set; }
+
+        public bool ToParameterizeServiceUrl { get; private set; }
+
+        public bool ToParameterizeNamedValue { get; private set; }
+
+        public bool ToParameterizeApiLoggerId { get; private set; }
+
+        public bool ToParameterizeLogResourceId { get; private set; }
+
+        public bool NotIncludeNamedValue { get; private set; }
+
+        public bool ParamNamedValuesKeyVaultSecrets { get; private set; }
+
+        public int OperationBatchSize { get; private set; }
+
+        public bool ToParameterizeBackend { get; set; }
```

Comment:
Can you make all the props have a private setter?

Reply:
done